From 1e7b5bc912c81766597975b8a9a3f8e512d8f520 Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Wed, 1 Jul 2026 13:57:54 -0400
Subject: [PATCH 2/9] stages/coreos.live-artifacts: drop separate
 make_efi_bootfile() function

It didn't quite have enough code to justify having it split out on its
own at this point.
---
 stages/org.osbuild.coreos.live-artifacts.mono | 58 +++++++++----------
 1 file changed, 26 insertions(+), 32 deletions(-)

diff --git a/stages/org.osbuild.coreos.live-artifacts.mono b/stages/org.osbuild.coreos.live-artifacts.mono
index 029af63b..5db23e59 100755
--- a/stages/org.osbuild.coreos.live-artifacts.mono
+++ b/stages/org.osbuild.coreos.live-artifacts.mono
@@ -171,33 +171,6 @@ def find_efi_vendor_dir_name(efidir):
     return vendor_id
 
 
-# This creates efiboot.img, which is a FAT filesystem.
-def make_efi_bootfile(loop_client, input_tarball, output_efiboot_img):
-    # Create the efiboot image file. Determine the size we should make
-    # it by taking the tarball size and adding 2MiB for fs overhead.
-    size = os.path.getsize(input_tarball) + 2 * 1024 * 1024
-    with open(output_efiboot_img, "wb") as out:
-        out.truncate(size)
-    # Make loopback device; mkfs; populate with files
-    with loop_client.device(output_efiboot_img) as loopdev:
-        # On RHEL 8, when booting from a disk device (rather than a CD),
-        # https://github.com/systemd/systemd/issues/14408 causes the
-        # hybrid ESP to race with the ISO9660 filesystem for the
-        # /dev/disk/by-label symlink unless the ESP has its own label,
-        # so set EFI-SYSTEM for consistency with the metal image.
-        # This should not be needed on Fedora or RHEL 9, but seems like
-        # a good thing to do anyway.
-        label = 'EFI-SYSTEM'
-        # NOTE: the arguments to mkfs here match how virt-make-fs calls mkfs
-        subprocess.check_call(['mkfs', '-t', 'vfat', '-I', '--mbr=n', '-n', label, loopdev])
-        with tempfile.TemporaryDirectory() as d:
-            try:
-                subprocess.check_call(['mount', '-o', 'utf8', loopdev, d])
-                subprocess.check_call(['tar', '-C', d, '-xf', input_tarball])
-            finally:
-                subprocess.check_call(['umount', d])
-
-
 def parse_metal_inputs(inputs):
     def get_filepath_from_input(name):
         files = inputs[name]["data"]["files"]
@@ -575,16 +548,37 @@ def mkefiboot(efidir, output_efiboot_img, loop_client):
             tarinfo.mode = 0o0644
         return tarinfo
 
-    # Install binaries from boot partition
+    # Install binaries from the efidir
     # Manually construct the tarball to ensure proper permissions and ownership
     efitarfile = tempfile.NamedTemporaryFile(suffix=".tar")
     with tarfile.open(efitarfile.name, "w:", dereference=True) as tar:
         tar.add(efidir, arcname="/EFI", filter=strip_tar)
 
-    # Create the efiboot.img file in the images/ dir
-    make_efi_bootfile(loop_client,
-                      input_tarball=efitarfile.name,
-                      output_efiboot_img=output_efiboot_img)
+    # Create the efiboot image file. Determine the size we should make
+    # it by taking the tarball size and adding 2MiB for fs overhead.
+    size = os.path.getsize(efitarfile.name) + 2 * 1024 * 1024
+    with open(output_efiboot_img, "wb") as out:
+        out.truncate(size)
+
+    # Make loopback device; mkfs; populate with files
+    with loop_client.device(output_efiboot_img) as loopdev:
+        # On RHEL 8, when booting from a disk device (rather than a CD),
+        # https://github.com/systemd/systemd/issues/14408 causes the
+        # hybrid ESP to race with the ISO9660 filesystem for the
+        # /dev/disk/by-label symlink unless the ESP has its own label,
+        # so set EFI-SYSTEM for consistency with the metal image.
+        # This should not be needed on Fedora or RHEL 9, but seems like
+        # a good thing to do anyway.
+        label = 'EFI-SYSTEM'
+        # NOTE: the arguments to mkfs here match how virt-make-fs calls mkfs
+        subprocess.check_call(['mkfs', '-t', 'vfat', '-I', '--mbr=n', '-n', label, loopdev])
+        with tempfile.TemporaryDirectory() as d:
+            try:
+                subprocess.check_call(['mount', '-o', 'utf8', loopdev, d])
+                subprocess.check_call(['tar', '-C', d, '-xf', efitarfile.name])
+            finally:
+                subprocess.check_call(['umount', d])
+
 
 def mkrootfs_metal(paths, workdir, img_metal, fstype, fsoptions, loop_client):
     """
-- 
2.54.0

